documentation/modules/configuring/proc-altering-connector-offsets.adoc

```diff
@@ -0,0 +1,127 @@
+The offset changes are specified in a config map. 
+You can reuse the same config map used to xref:proc-listing-connector-offsets-{context}[list offsets].
+The offset alteration is applied by annotating the `KafkaConnector` resource.
+
+In this procedure, we alter the offset position for a source connector named `my-source-connector`.
+
+NOTE: You can also alter connector offsets using the `PATCH /connectors/{connector}/offsets` request through the Kafka Connect Rest API.
+
+.Prerequisites
+
+* The Cluster Operator is running.
+
+.Procedure
+
+. Edit the `KafkaConnector` resource to stop the connector and include the `listOffsets` configuration.
++
+.Example configuration to stop a connector and alter offsets
+[source,yaml,subs="+attributes"]
+----
+apiVersion: {KafkaConnectorApiVersion}
+kind: KafkaConnector
+metadata:
+  name: my-source-connector
+  labels:
+    strimzi.io/cluster: my-connect-cluster
+spec:
+  state: stopped # <1>
+  alterOffsets:
+    fromConfigMap: # <2>
+      name: my-connector-offsets # <3>
+  # ... 
+----
+<1> Changes the state of the connector to `stopped`. The default state for the connector when this property is not set is `running`.
+<2> The reference to the config map that provides the update. 
+<3> The name of the config map, which is named `my-connector-offsets` in this example. 
+
+. Edit the config map to make the alteration.
++
+In this example, we're resetting the offset position for a source connector to 15000.
++
+.Example source connector offset list configuration
+[source,yaml,subs="+attributes"]
+----
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  # ...
+data: 
+  offsets.json: |-
+    {
+      "offsets" : [ {
+        "partition" : {
+          "filename" : "/data/myfile.txt"
+        },
+        "offset" : {
+          "position" : 15000 # <1>
+        }
+      } ]
+    } 
+----
+<1> The updated offset position in the source partition.
+
+. Run the command to update the offset position by annotating the `KafkaConnector` resource:
++
+[source,shell]
+----
+kubectl annotate kafkaconnector my-source-connector strimzi.io/connector-offsets=alter -n <namespace>
+----
++
+The annotation remains until either the update operation succeeds, or it is manually removed from the resource.
+
+. Inspect the contents of the config map to verify the update to the offset position:
++
+[source,shell]
+----
+kubectl describe configmap my-connector-offsets -n <namespace>
+----
++
+Strimzi puts the offsets into the `offsets.json` property. 
+It doesn't overwrite any other properties when updating an existing config map.
++
+.Example source connector offset list
+[source,yaml,subs="+attributes"]
+----
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  # ...
+data: 
+  offsets.json: |-
+    {
+      "offsets" : [ {
+        "partition" : {
+          "filename" : "/data/myfile.txt"
+        },
+        "offset" : {
+          "position" : 15000
+        }
+      } ]
+    } 
+----
```

Comment:
Looks like some copy-paste leftover from listing the offsets?

```suggestion
```

Reply:
I thought we should include a verify step for all three procedures.

Reply:
Ok ... but for that it is missing the list annotation, or? The only way to verify it would be to list the offsets again after the alter annotation is removed.